tests/test_group_layer_widget.py

```diff
@@ -66,3 +66,49 @@ def test_double_click_edit(group_layer_widget, double_click_on_view):
     node_index = group_layers_model.index(0, 0)
     double_click_on_view(group_layers_view, node_index)
     assert group_layers_view.state() == group_layers_view.EditingState
+
+
+def test_layer_sync(make_napari_viewer, image_layer, points_layer):
+    """
+    Test the synchronisation between the widget and the main LayerList.
+    This test will be redundant when the plugin functionality replaces the
+    main viewer functionality.
+    """
+    viewer = make_napari_viewer()
+    viewer.add_layer(image_layer)
+
+    widget = GroupLayerWidget(viewer)
+    assert len(widget.group_layers) == 1
+
+    # Check that adding a layer to the viewer means it is also added to
+    # the group layers view.
+
+    viewer.add_layer(points_layer)
+    assert len(widget.group_layers) == 2
+
+    # Check that reordering the layer order in the group layers view
+    # forces an update in the main viewer.
+    # However, don't forget that the ordering is REVERSED in the GUI, so
+    # viewer.layers[-1] == widget.group_layers[0], etc.
+    for in_viewer, in_widget in zip(
+        reversed(viewer.layers),
+        [node.layer for node in widget.group_layers],
+        strict=True,
```

Comment:
I think `strict` was only introduced in 3.10, so you'll probably have to remove it to get the tests to pass with 3.9